Doc/library/datetime.rst

```diff
@@ -2139,13 +2160,18 @@ Notes:
    For an aware object:
 
    ``%z``
-      :meth:`utcoffset` is transformed into a 5-character string of the form
-      +HHMM or -HHMM, where HH is a 2-digit string giving the number of UTC
+      :meth:`utcoffset` is transformed into a string of the form
+      ±HHMM[SS], where HH is a 2-digit string giving the number of UTC
       offset hours, and MM is a 2-digit string giving the number of UTC offset
+      minutes, and SS is a string giving the number of UTC offset
+      seconds.  The SS part is omitted when the offset is a whole number of
       minutes.  For example, if :meth:`utcoffset` returns
       ``timedelta(hours=-3, minutes=-30)``, ``%z`` is replaced with the string
       ``'-0330'``.
 
+   .. versionchanged:: 3.7
+      The UTC offset is not restricted to a whole number of minutes.
```

Comment:
I suggest to be more explicit: mention that optional support for seconds was added, since microseconds are not supported here.